logger/src/lib.rs

```diff
@@ -676,20 +675,27 @@ impl Logger {
 
         set_max_level(Level::Trace.to_level_filter());
 
+        self.log_helper(
+            format!("Running {} v{}", app_info.name, app_info.version),
+            Some(Destination::Pipe),
+        );
+        LOGGER.level_info.set_writer(Destination::Pipe);
         STATE.store(INITIALIZED, Ordering::SeqCst);
 
-        self.log_helper(format!("Running {} v{}", app_info.name, app_info.version));
-
         Ok(())
     }
 
     // In a future PR we'll update the way things are written to the selected destination to avoid
     // the creation and allocation of unnecessary intermediate Strings. The log_helper method takes
     // care of the common logic involved in both writing regular log messages, and dumping metrics.
-    fn log_helper(&self, msg: String) {
+    fn log_helper(&self, msg: String, forced_destination: Option<Destination>) {
+        let mut destination = self.level_info.writer();
```

Comment:
Just saying, I think this looks nicer as:
```rust
let destination = forced_destination.unwrap_or(self.level_info.writer());
```
In fact, you can even combine this with the following `match`, so you can have something like (though the borrow checker might complain here):
```rust
match forced_destination.unwrap_or(self.level_info.writer()) {
...
}
```
Also, it's interesting to check out the difference between `Option::unwrap_or()` and `Option::unwrap_or_else()`.